File: jgBay/search/src/main/java/com/jahnelgroup/jgbay/search/data/auction/Auction.java
package com.jahnelgroup.jgbay.search.data.auction;

import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.elasticsearch.annotations.Document;

import java.util.HashSet;
import java.util.Set;

@Data
@Document(indexName = "auctions", refreshInterval = "-1")
public class Auction {

    @Id
    private Long id;
    private String status;
    private String title;
    private Long sellerId;
    private Short numberOfBids;
    private Set<String> categories = new HashSet<>();

}

File: jgBay/search/src/main/kotlin/com/jahnelgroup/jgbay/search/data/auction/Auction.kt
package com.jahnelgroup.jgbay.search.data.auction

import org.springframework.data.annotation.Id
import org.springframework.data.elasticsearch.annotations.Document

@Document(indexName = "auctions", refreshInterval = "-1")
data class Auction(
    @Id
    var id : Long = 0,
    var status: String = "",
    var title: String = "",
    var seller: Long = 0,
    var numberOfBids: Int = 0,
    var categories: Set<String> = emptySet()
)